operators/gstreamer/CMakeLists.txt

```
@@ -0,0 +1,142 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.20)
+project(holoscan_gstreamer_bridge)
+
+# Find required packages
+find_package(PkgConfig REQUIRED)
+
+# Find Holoscan
+find_package(holoscan REQUIRED CONFIG
+             PATHS "/opt/nvidia/holoscan" "/workspace/holoscan-sdk/install")
+
+# Find GStreamer
+pkg_check_modules(GSTREAMER REQUIRED gstreamer-1.0)
+pkg_check_modules(GSTREAMER_BASE REQUIRED gstreamer-base-1.0)
+pkg_check_modules(GSTREAMER_VIDEO REQUIRED gstreamer-video-1.0)
+pkg_check_modules(GSTREAMER_CUDA gstreamer-cuda-1.0)  # Optional - only available in GStreamer 1.24+
+pkg_check_modules(GSTREAMER_APP REQUIRED gstreamer-app-1.0)
+
+# Create the Holoscan GStreamer Bridge library
+add_library(holoscan_gstreamer_bridge SHARED
+  gst/buffer.cpp
+  gst/caps.cpp
+  gst/memory.cpp
+  gst/message.cpp
+  gst/video_info.cpp
+  gst_src_bridge.cpp
+  gst_video_recorder_operator.cpp
+)
+
+# Create alias for consistent naming
+add_library(holoscan::ops::gstreamer_bridge ALIAS holoscan_gstreamer_bridge)
+
+# Set target properties
+set_target_properties(holoscan_gstreamer_bridge PROPERTIES
+  C_STANDARD 99
+  C_STANDARD_REQUIRED ON
+  CXX_STANDARD 17
+  CXX_STANDARD_REQUIRED ON
+  POSITION_INDEPENDENT_CODE ON
+)
+
+# Set include directories (PUBLIC so they propagate to applications)
+target_include_directories(holoscan_gstreamer_bridge
+    PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:include>
+    ${GSTREAMER_INCLUDE_DIRS}
+    ${GSTREAMER_BASE_INCLUDE_DIRS}
+    ${GSTREAMER_VIDEO_INCLUDE_DIRS}
+    ${GSTREAMER_APP_INCLUDE_DIRS}
```

Recommend BUILD_INTERFACE for all, I'm assuming `GSTREAMER_INCLUDE_DIRS` are absolute paths in the build environment, please correct me if I'm wrong

```suggestion
    $<BUILD_INTERFACE:${GSTREAMER_INCLUDE_DIRS}>
    $<BUILD_INTERFACE:${GSTREAMER_BASE_INCLUDE_DIRS}>
    $<BUILD_INTERFACE:${GSTREAMER_VIDEO_INCLUDE_DIRS}>
    $<BUILD_INTERFACE:${GSTREAMER_APP_INCLUDE_DIRS}>
```

---

You're correct that `GSTREAMER_INCLUDE_DIRS` are absolute paths from pkg-config. However, I intentionally don't use `BUILD_INTERFACE` here because:
1. Our public headers (`gst_video_recorder_operator.hpp`, `gst_src_bridge.hpp`) expose GStreamer types (e.g., `GstElement*`, `GstBuffer*`)
2. Consuming applications need these include paths to compile against our headers
3. GStreamer is a system dependency (found via pkg-config), not a bundled library, so the paths are consistent between build and install

Using BUILD_INTERFACE would require all consuming applications to also call `pkg_check_modules(GSTREAMER ...)`, which breaks encapsulation. The current approach is standard practice for CMake libraries wrapping system dependencies.